basic-algorithms.js

```diff
@@ -1,7 +1,71 @@
 // Names and Input
+let hacker1 = "Cristina";
+console.log(`The driver's name is ${hacker1}`);
 
+let hacker2 = prompt("Which is the navigator's name?");
+console.log(`The navigator's name is ${hacker2}`);
 
 //Conditionals
-
+if (hacker1.length > hacker2.length) {
+  console.log(
+    `The Driver ${hacker1} has the longest name, it has ${
+      hacker1.length
+    } characters`
+  );
+} else if (hacker2.lenght > hacker1.length) {
+  console.log(
+    `Yo, navigator ${hacker2} got the longest name, it has ${
+      hacker2.length
+    } characters`
+  );
+} else if (hacker1.length === hacker2.length) {
+  console.log(
+    `wow, you both ${hacker1} and ${hacker2} got equally long names, ${
+      hacker1.length
+    } characters!!`
+  );
+}
 
 // Lorem ipsum generator
+let driver = hacker1;
+let navigator = hacker2;
+
+for (i = 0; i < driver.length; i++) {
+  console.log(driver[i].toUpperCase());
+  i += " ";
```

Comment:
Esto no lo hagas nunca, dentro de un bucle For, no debes modificar el valor de la variable que usa para contar iteraciones, el propio bucle es el encargado de actualizarla a cada vuelta. Además cuando haces i+= " ", estás convirtiendo un número a string.
Te funciona porque internamente cuando el bucle actualiza el valor de i, hace un autoboxing a number.

Reply:
gracias jorge, aquí confundí el valor del iterator i con el de la variable driver. ya lo estoy corrigiendo en el código.